diff -up qtsmbstatus-2.1.2/server/etc/qtsmbstatusd.init qtsmbstatus-2.1.2/server/etc/qtsmbstatusd
--- qtsmbstatus-2.1.2/server/etc/qtsmbstatusd.init	2009-04-12 14:46:48.000000000 +0200
+++ qtsmbstatus-2.1.2/server/etc/qtsmbstatusd	2009-04-12 17:51:37.000000000 +0200
@@ -1,8 +1,11 @@
 #!/bin/sh
-#  script for qtsmbstatusd
 #
+# qtsmbstatusd
+#
+# chkconfig:   35 99 20
+# description: qtsmbstatusd server (QtSmbstatus is a graphical user \
+#              interface for smbstatus)
 
-# For LSB
 ### BEGIN INIT INFO
 # Provides:          qtsmbstatusd
 # Required-Start:    $samba
@@ -13,69 +16,99 @@
 # Description:       QtSmbstatus is a graphical user interface for smbstatus
 ### END INIT INFO
 
-# For Mandrake/Mandriva/Fedora:
-# chkconfig: 35 99 20
-# description: qtsmbstatusd server (QtSmbstatus is a graphical user interface for smbstatus)
+# Source function library.
+. /etc/rc.d/init.d/functions
+
+exec="/usr/bin/qtsmbstatusd"
+prog="qtsmbstatusd"
+config="/etc/qtsmbstatusd/qtsmbstatusd.conf"
 
-PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:$PATH
-NAME=qtsmbstatusd
+[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
 
-# see if qtsmbstatusd is running
-pid=`ps ax | awk '{print $1,$5}' | grep $NAME | awk '{print $1}' | awk '{print $1}'`
+lockfile=/var/lock/subsys/$prog
 
 # get locale
 default_locale=en_US.UTF-8
-if [ -f /etc/default/locale ]; then
-        # Debian systems
-        system_locale=`grep -re "LANG=" /etc/default/locale | awk 'BEGIN { FS = "[\"]" } { print $2 }'`
-elif [ -f /etc/sysconfig/language ]; then
-        # SuSE systems
-        system_locale=`grep -re "LANG=" /etc/sysconfig/language | awk 'BEGIN { FS = "[\"]" } { print $2 }'`
-elif [ -f /etc/sysconfig/i18n ]; then
+if [ -f /etc/sysconfig/i18n ]; then
         # RedHat/mandriva systems
         system_locale=`grep -re "LANG=" /etc/sysconfig/i18n | awk 'BEGIN { FS = "[\"]" } { print $2 }'`
-elif [ -f /etc/profile ]; then
-	#default
-        system_locale=`grep -re "LANG=" /etc/profile | awk 'BEGIN { FS = "[\"]" } { print $2 }'`
 fi
 
 if [ -z $system_locale ]; then system_locale=$default_locale; fi
 
-# Start or stop the qtsmbstatusd server
-case $1 in
-	start | restart | reload | force-reload)
-		export LANG=$system_locale
-		if test "$pid" != ""; then
-			kill $pid
-			if qtsmbstatusd -d; then
-			echo "qtsmbstatusd server restarted."
-			fi
-                else
-			if qtsmbstatusd -d; then
-			echo "qtsmbstatusd server started."
-			fi
-		fi
-                ;;
-	stop)
-		if test "$pid" != ""; then
-		  kill -3 $pid
-		fi
-		echo "qtsmbstatusd server stopped"
-		;;
-
-	status)
-		if test "$pid" != ""; then
-			echo "qtsmbstatusd : server is running."
-		else
-			echo "qtsmbstatusd : server is not running."
-		fi
-		;;
-
-	*)
-		echo "Usage: qtsmbstatusd {start|stop|reload|restart|status|force-reload}"
-		exit 1
-		;;
-esac
-
-exit 0
 
+start() {
+    [ -x $exec ] || exit 5
+    [ -f $config ] || exit 6
+    export LANG=$system_locale
+    echo -n $"Starting $prog: "
+    $exec -d
+    retval=$?
+    echo
+    [ $retval -eq 0 ] && touch $lockfile
+    return $retval
+}
+
+stop() {
+    echo -n $"Stopping $prog: "
+    killproc $prog
+    retval=$?
+    echo
+    [ $retval -eq 0 ] && rm -f $lockfile
+    return $retval
+}
+
+restart() {
+    stop
+    start
+}
+
+reload() {
+    restart
+}
+
+force_reload() {
+    restart
+}
+
+rh_status() {
+    # run checks to determine if the service is running or use generic status
+    status $prog
+}
+
+rh_status_q() {
+    rh_status >/dev/null 2>&1
+}
+
+
+case "$1" in
+    start)
+        rh_status_q && exit 0
+        $1
+        ;;
+    stop)
+        rh_status_q || exit 0
+        $1
+        ;;
+    restart)
+        $1
+        ;;
+    reload)
+        rh_status_q || exit 7
+        $1
+        ;;
+    force-reload)
+        force_reload
+        ;;
+    status)
+        rh_status
+        ;;
+    condrestart|try-restart)
+        rh_status_q || exit 0
+        restart
+        ;;
+    *)
+        echo $"Usage: $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
+        exit 2
+esac
+exit $?
