diff -up qtsmbstatus-2.0.5/server/etc/qtsmbstatusd.init qtsmbstatus-2.0.5/server/etc/qtsmbstatusd
--- qtsmbstatus-2.0.5/server/etc/qtsmbstatusd.init	2008-08-28 17:41:46.000000000 +0200
+++ qtsmbstatus-2.0.5/server/etc/qtsmbstatusd	2008-09-04 22:55:13.000000000 +0200
@@ -1,8 +1,11 @@
 #!/bin/sh
-#  script for qtsmbstatusd
 #
+# qtsmbstatusd
+#
+# chkconfig:   35 99 20
+# description: qtsmbstatusd server (QtSmbstatus is a graphical user \
+#              interface for smbstatus)
 
-# For LSB
 ### BEGIN INIT INFO
 # Provides:          qtsmbstatusd
 # Required-Start:    $samba
@@ -13,49 +16,88 @@
 # Description:       QtSmbstatus is a graphical user interface for smbstatus
 ### END INIT INFO
 
-# For Mandrake/Mandriva/Fedora:
-# chkconfig: 35 99 20
-# description: qtsmbstatusd server (QtSmbstatus is a graphical user interface for smbstatus)
-
-
-NAME=qtsmbstatusd
-
-# see if qtsmbstatusd is running
-pid=`ps ax | awk '{print $1,$5}' | grep $NAME | awk '{print $1}' | awk '{print $1}'`
-
-# Start or stop the qtsmbstatusd server
-case $1 in
-	start | restart | reload | force-reload)
-                if test "$pid" != ""; then
-                        kill $pid
-			if /usr/bin/qtsmbstatusd -d; then
-		  	echo "qtsmbstatusd server restarted."
-			fi
-                else
-                  	if /usr/bin/qtsmbstatusd -d; then
-		  	echo "qtsmbstatusd server started."
-			fi
-                fi
-                ;;
-	stop)
-		if test "$pid" != ""; then
-		  kill -3 $pid
-		fi
-		echo "qtsmbstatusd server stopped"
-		;;
-
-	status)
-		if test "$pid" != ""; then
-			echo "qtsmbstatusd : server is running."
-		else
-			echo "qtsmbstatusd : server is not running."
-		fi
-		;;
-
-	*)
-		echo "Usage: qtsmbstatusd {start|stop|reload|restart|status|force-reload}"
-		exit 1
-		;;
-esac
+# Source function library.
+. /etc/rc.d/init.d/functions
 
-exit 0
+exec="/usr/bin/qtsmbstatusd"
+prog="qtsmbstatusd"
+config="/etc/qtsmbstatusd/qtsmbstatusd.conf"
+
+[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
+
+lockfile=/var/lock/subsys/$prog
+
+start() {
+    [ -x $exec ] || exit 5
+    [ -f $config ] || exit 6
+    echo -n $"Starting $prog: "
+    $exec -d
+    retval=$?
+    echo
+    [ $retval -eq 0 ] && touch $lockfile
+    return $retval
+}
+
+stop() {
+    echo -n $"Stopping $prog: "
+    killproc $prog
+    retval=$?
+    echo
+    [ $retval -eq 0 ] && rm -f $lockfile
+    return $retval
+}
+
+restart() {
+    stop
+    start
+}
+
+reload() {
+    restart
+}
+
+force_reload() {
+    restart
+}
+
+rh_status() {
+    # run checks to determine if the service is running or use generic status
+    status $prog
+}
+
+rh_status_q() {
+    rh_status >/dev/null 2>&1
+}
+
+
+case "$1" in
+    start)
+        rh_status_q && exit 0
+        $1
+        ;;
+    stop)
+        rh_status_q || exit 0
+        $1
+        ;;
+    restart)
+        $1
+        ;;
+    reload)
+        rh_status_q || exit 7
+        $1
+        ;;
+    force-reload)
+        force_reload
+        ;;
+    status)
+        rh_status
+        ;;
+    condrestart|try-restart)
+        rh_status_q || exit 0
+        restart
+        ;;
+    *)
+        echo $"Usage: $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
+        exit 2
+esac
+exit $?
